src/ObjectMakr.ts

```diff
@@ -62,8 +62,7 @@ export class ObjectMakr implements IObjectMakr {
         this.doPropertiesFull = settings.doPropertiesFull;
         this.indexMap = settings.indexMap;
         this.onMake = settings.onMake;
-
-        this.functions = {};
+        this.functions = settings.functions || {};
```

Comment:
It would be better to make a copy of the settings object. This way we don't accidentally modify other people's / scopes' objects.